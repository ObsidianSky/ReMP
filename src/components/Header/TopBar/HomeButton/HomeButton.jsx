import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import { navigateHome } from '../../../../actions/navigation.actions';
import Button from '../../../common/Button/Button';

const HomeButton = ({ navigateHome }) => (
    <Button onClick={navigateHome} className="btn--wide">Search</Button>
);

HomeButton.propTypes = {
    navigateHome: PropTypes.func.isRequired
};

export default connect(null, { navigateHome })(HomeButton);
