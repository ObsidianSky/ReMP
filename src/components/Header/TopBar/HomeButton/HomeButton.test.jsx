import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { HomeButton } from './HomeButton';

describe('HomeButton', () => {
    let props;
    let navigateHome;

    beforeEach(() => {
        navigateHome = jest.fn();
        props = { navigateHome };
    });

    test('should renders correctly', () => {
        const wrapper = shallow(<HomeButton { ...props }/>);

        expect(toJson(wrapper)).toMatchSnapshot();
    });
});

