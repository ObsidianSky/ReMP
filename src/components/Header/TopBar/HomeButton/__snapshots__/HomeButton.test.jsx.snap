// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeButton should renders correctly 1`] = `
<Button
  className="btn--wide"
  onClick={[Function]}
  text=""
>
  Search
</Button>
`;
